### PR TITLE
Bug 1205257 - Spark distro does not rebuild fine from build.sh

### PR DESCRIPTION
**Pull request by rickychien**
https://bugzilla.mozilla.org/show_bug.cgi?id=1205257

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=debf13aa3c3fbddab38abda1fdd8f80c44b67d7d

File: build/preload.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ var python = new utils.Commander('python');
 
 exports.execute = function(opts) {
   var GAIA_DISTRIBUTION_DIR = utils.getEnv('GAIA_DISTRIBUTION_DIR');
+
+  if (GAIA_DISTRIBUTION_DIR.indexOf('distros/spark') === -1) {
```
Comment (lissyx) [inline build/preload.js]: That feels ... Strange ? Why ?

Comment (julienw) [inline build/preload.js]: yeah I think we'll want other distros to work in the future :)

Comment (julienw) [inline build/preload.js]: I think you merely want to see if GAIA_DISTRIBUTION_DIR is not empty.

File: Makefile [hunk 2]
```diff
@@ -616,6 +615,10 @@ webapp-optimize: app
 .PHONY: webapp-zip
 webapp-zip: app
 
+.PHONY: preload-app
+preload-app:
+	@$(call $(BUILD_RUNNER),preload,GAIA_DIR="$(GAIA_DIR)" GAIA_DISTRIBUTION_DIR="$(GAIA_DISTRIBUTION_DIR)")
```
Comment (julienw) [inline Makefile]: I don't really like using "@" because we don't see the command being run

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9c2a4c673429e894e0627fb6c220cc6e552c026d

File: build/preload.js [hunk 3]
```diff
@@ -5,6 +5,11 @@ var python = new utils.Commander('python');
 
 exports.execute = function(opts) {
   var GAIA_DISTRIBUTION_DIR = utils.getEnv('GAIA_DISTRIBUTION_DIR');
+
+  if (GAIA_DISTRIBUTION_DIR === '') {
```
Comment (lissyx) [inline build/preload.js]: Maybe you could even check if that directory exists and is accessible?

**Pull request merged**